test/unit/services/test_image.py

```diff
@@ -247,3 +247,45 @@ def test_list(self):
         self.mock_list()
         self.call_list(entity_id=str(ObjectId()), primary=False)
         self.check_list_success()
+
+
+class DeleteDSL(ImageServiceDSL):
+    """Base class for `delete` tests."""
+
+    _delete_image_id: str
+    _expected_image_out: ImageOut
+    _delete_image_id: str
+    _delete_image_object_key: str
+
+    def mock_delete(self, image_in_data: dict) -> None:
+        """
+        Mocks repo methods appropriately to test the `delete` service method.
+
+        :param image_in_data: Dictionary containing the image data as would be required for an `ImageIn`
+                                   database model (i.e. no created and modified times required).
+        """
+        self._expected_image_out = ImageOut(**ImageIn(**image_in_data).model_dump())
+        self.mock_image_repository.get.return_value = self._expected_image_out
+        self._delete_image_id = self._expected_image_out.id
+        self._delete_image_object_key = self._expected_image_out.object_key
+
+    def call_delete(self) -> None:
+        """Calls the `ImageService` `delete` method."""
+
+        self.image_service.delete(self._delete_image_id)
+
+    def check_delete_success(self) -> None:
+        """Checks that a prior call to `call_delete` worked as expected."""
+
+        self.mock_image_repository.delete.assert_called_once_with(self._delete_image_id)
+        self.mock_image_store.delete.assert_called_once_with(self._delete_image_object_key)
```

Comment:
```suggestion
        self.mock_image_store.delete.assert_called_once_with(self._delete_image_object_key)
        self.mock_image_repository.delete.assert_called_once_with(self._delete_image_id)
```
Just to match the call order.